### PR TITLE
Bug fix in Sagemaker plugin

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
_Please replace this text with a description of what this PR accomplishes._

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Comment by bnsblue** [comment]
🔨

File: go/tasks/plugins/k8s/sagemaker/sagemaker.go [hunk 1]
```diff
@@ -78,10 +78,16 @@ func (m awsSagemakerPlugin) BuildResourceForTrainingJob(
...
 		return nil, errors.Errorf("Required input not specified: [train]")
 	}
+	if trainPathLiteral.GetScalar() == nil || trainPathLiteral.GetScalar().GetBlob() == nil {
+		return nil, errors.Errorf("[train] Input is required and should be of Type [Scalar.Blob]")
+	}
 	validatePathLiteral, ok := inputLiterals["validation"]
 	if !ok {
 		return nil, errors.Errorf("Required input not specified: [validation]")
 	}
+	if validatePathLiteral.GetScalar() == nil || validatePathLiteral.GetScalar().GetBlob() == nil {
```
Comment (kumare3) [inline go/tasks/plugins/k8s/sagemaker/sagemaker.go:88]: @bnsblue can we find more cases in which we have a potential NPE and fix them before we merge?

Comment (bnsblue) [reply go/tasks/plugins/k8s/sagemaker/sagemaker.go:88]: sure

Comment (EngHabu) [reply go/tasks/plugins/k8s/sagemaker/sagemaker.go:88]: If you haven't, please add unit tests to these functions to cover branches... aim for 80%+ code coverage if we aren't there yet.. for this package at least

Comment (bnsblue) [reply go/tasks/plugins/k8s/sagemaker/sagemaker.go:88]: ok then we should probably merge https://github.com/lyft/flytepropeller-private/pull/257/files alone first

File: go/tasks/plugins/k8s/sagemaker/config/config.go [hunk 2]
```diff
@@ -19,8 +19,8 @@ var (
 						Region: "us-east-1",
 						VersionConfigs: []VersionConfig{
 							{
-								Version: "0.91",
-								Image:   "811284229777.dkr.ecr.us-east-1.amazonaws.com/xgboost:latest",
+								Version: "0.90",
```
Comment (kumare3) [inline go/tasks/plugins/k8s/sagemaker/config/config.go:22]: i think in the future we should have defaults for all regions?

**Comment by bnsblue** [comment]
This PR is merged in order to fix an important bug in a timely manner. Extra unit tests are added but we haven't reached the coverage goal. Will try to reach the goal in a separate PR.

**Pull request merged**